### ISSUE TITLE
Revert usage of forwardRef in version ^8 and re-release as a breaking change

### ISSUE THREAD

**Issue by jedwards1211** [cssinjs/react-jss#255]
`injectSheet(styles)(MyComponent)` is returning a non-component that I can't wrap in  react-redux' `connect`. Argh!
```
const pleaseBeMoreCareful = injectSheet(styles)(SitePlanningSidebar)
console.log(pleaseBeMoreCareful)
{
  $$typeof: Symbol(react.forward_ref),
  InnerComponent: ƒ SitePlanningSidebar(_ref),
  render: ƒ forwardRef(props, ref),
}
```

**Comment by jedwards1211** [comment]
See https://github.com/ReactTraining/react-router/issues/6056, they were more careful about this

**Comment by jedwards1211** [comment]
https://github.com/reduxjs/react-redux/issues/914

**Comment by kof** [comment]
Lets fix this instead of removing support https://github.com/reduxjs/react-redux/issues/914#issuecomment-377406014

**Comment by jedwards1211** [comment]
From @gaearon: 

> Your app broke because you updated a component library (not React!) to a version that includes a breaking change (using forwardRef). Note that React adding forwardRef is not a breaking change; a component library using forwardRef is.

This should have been released as a breaking change, not a minor version bump.

**Comment by jedwards1211** [comment]
I would say revert the change and release as `8.6.0` and then re-release `8.5.0` as `9.0.0`.

**Comment by gaearon** [comment]
Yeah, I agree it should be a semver-major change because the observable behavior is different.

**Comment by kof** [comment]
Whats the point of having forwardRef support in react-jss at all and release a major version of it?

**Comment by gaearon** [comment]
It lets people get refs to the underlying components.

**Comment by kof** [comment]
yes, I mean in our case we have innerRef prop which does that, forwardRef api seems more like a user-land one

**Comment by kof** [comment]
@jedwards1211 8.5.1 released cc @HenriBeck

**Comment by jedwards1211** [comment]
Thanks @kof, I kind of prefer the simplicity of `innerRef` as well.  Would you be open to a PR to add a render props version of `injectSheet` to this lib?  I've been wanting one for awhile and it wouldn't suffer from this problem.  For example:

```js
import {createStyled} from 'react-jss'

const styles = {
  root: {
    backgroundColor: 'red'
  }
}

const MyCompStyles = createStyled(styles)

const MyComp = props => (
  <MyCompStyles>
    {({classes}) => (
      <div ref={c => console.log(c)} className={classes.root} />
    )}
  </MyCompStyles>
)
```

**Comment by gaearon** [comment]
>yes, I mean in our case we have innerRef prop which does that, forwardRef api seems more like a user-land one

The intention *is* for third-party HOCs to use `forwardRef` so people don't have to learn a different naming convention for each library, and so that HOCs compose together without "agreeing" on the prop name.

**Comment by kof** [comment]
@jedwards1211 separate issue please, yes if we find a way (not so easy)

**Comment by kof** [comment]
Now rereleasing with forwardRef as-is still doesn't fix it when used with react-redux. I haven't checked yet what needs to be done there, therefore a new issue #256

**Comment by gaearon** [comment]
@kof 

>Now rereleasing with forwardRef as-is still doesn't fix it when used with react-redux

That's their problem and on them to fix.

**Comment by kof** [comment]
Yes, but even if they fix, there will be many people using older react-redux version and complaining it doesn't work. Same for probably many other popular libraries. It is like breaking compatibility with half of the world.

**Comment by gaearon** [comment]
I don't this realistically a lot of other libraries do that. Most HOCs I've seen don't do any validation of inputs.

React Redux can release a patch for both actively used majors (4.x and 5.x) so people can always update it.

We've relied on that even more significantly already. For example, the new `getDerivedStateFromProps` lifecycle is broken for people with older React Redux versions, and yet after we released patches for 4.x and 5.x, people stopped complaining.
